public/components/event_analytics/utils/__tests__/utils.test.tsx

```diff
@@ -114,4 +116,27 @@ describe('Utils event analytics helper functions', () => {
       values: [0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 5, 4, 2, 3, 3, 1, 0, 0],
     });
   });
+
+  it('validates redoQuery function', () => {
+    expect(
+      redoQuery(
+        'now/y',
+        'now',
+        "source = opensearch_dashboards_sample_data_logs | where match(request,'filebeat')",
+        'timestamp',
+        {
+          current: [
+            {
+              id: 'timestamp',
+              direction: 'asc',
+            },
+          ],
+        },
+        {
+          current: [0, 100],
+        },
+        () => {}
+      )
+    ).toBe(undefined);
```

Comment:
i'm not sure how is expecting undefined validating the function?

you can probably at least do this
```suggestion
      const getEvents = jest.fn();
      redoQuery(
        'now/y',
        'now',
        "source = opensearch_dashboards_sample_data_logs | where match(request,'filebeat')",
        'timestamp',
        {
          current: [
            {
              id: 'timestamp',
              direction: 'asc',
            },
          ],
        },
        {
          current: [0, 100],
        },
        getEvents
      )
      expect(getEvents).toBeCalledWith(xxx)
```

Reply:
I didn't think to have jest test for a dummy getEvents, this seems like it would test the function well. Thanks for suggesting this, I've implemented it.